diff --git a/src/gpu/intel/matmul/grouped_post_ops_gen.cpp b/src/gpu/intel/matmul/grouped_post_ops_gen.cpp
index 65789b44af..579b0eca8c 100644
--- a/src/gpu/intel/matmul/grouped_post_ops_gen.cpp
+++ b/src/gpu/intel/matmul/grouped_post_ops_gen.cpp
@@ -157,8 +157,6 @@ std::string generate_post_ops_microgemm_header(
                << ";\n";
             os << "    tile_convert((*c_tile), eltwise_f16_boundary_" << i
                << ", convert_half);\n";
-            os << "    tile_convert(eltwise_f16_boundary_" << i
-               << ", (*c_tile), convert_float);\n";
             os << "    _Pragma(\"unroll\") for (int tile_i_" << i
                << " = 0; tile_i_" << i
                << " < sizeof((*c_tile).x) / sizeof((*c_tile).x[0]); "
@@ -169,9 +167,11 @@ std::string generate_post_ops_microgemm_header(
                << " < sizeof((*c_tile).x[0]) / "
                   "sizeof((*c_tile).x[0][0]); ++tile_s_"
                << i << ") {\n";
-            os << "            volatile float eltwise_gate_" << i
-               << " = (*c_tile).x[tile_i_" << i << "][tile_s_" << i
-               << "];\n";
+            os << "            volatile half eltwise_gate_f16_" << i
+               << " = eltwise_f16_boundary_" << i << ".x[tile_i_" << i
+               << "][tile_s_" << i << "];\n";
+            os << "            volatile float eltwise_gate_" << i
+               << " = convert_float(eltwise_gate_f16_" << i << ");\n";
             os << "            volatile float eltwise_value_" << i
                << " = eltwise_gate_" << i << " / (1.0f + native_exp(-("
                << to_ocl_float(e.eltwise.alpha) << ") * eltwise_gate_" << i
