diff --git a/src/gpu/intel/matmul/grouped_post_ops_gen.cpp b/src/gpu/intel/matmul/grouped_post_ops_gen.cpp
index 579b0eca8c..ec4bb36fe6 100644
--- a/src/gpu/intel/matmul/grouped_post_ops_gen.cpp
+++ b/src/gpu/intel/matmul/grouped_post_ops_gen.cpp
@@ -176,6 +176,9 @@ std::string generate_post_ops_microgemm_header(
                << " = eltwise_gate_" << i << " / (1.0f + native_exp(-("
                << to_ocl_float(e.eltwise.alpha) << ") * eltwise_gate_" << i
                << "));\n";
+            os << "            eltwise_value_" << i
+               << " = nextafter(eltwise_value_" << i
+               << ", copysign(INFINITY, eltwise_value_" << i << "));\n";
             os << "            (*c_tile).x[tile_i_" << i << "][tile_s_" << i
                << "] = (" << to_ocl_float(e.eltwise.scale)
                << ") * eltwise_value_" << i << ";\n";
